test/Spec.hs

```diff
@@ -0,0 +1,16 @@
+module Main
+    ( main
+    ) where
+
+-- import qualified Spec.Model
+import qualified Spec.Trace
+import           Test.Tasty
+
+main :: IO ()
+main = defaultMain tests
+
+tests :: TestTree
+tests = testGroup "trace testing"
+    [ Spec.Trace.tests
+    --, Spec.Model.tests
+    ]
```

Comment:
I've not used a lot of Tasty, but you should be able to have it automatically discover tests for you instead of having to declare tests by yourself with something like:

```
{-# OPTIONS_GHC -F -pgmF tasty-discover #-}
```

Reply:
This is super interesting! I would need to do a bit more reading to understand exactly what is going on there but I will definitely make sure to look into it :)